### ISSUE TITLE
转换的模型校验失败

### ISSUE THREAD

**Issue by xiaoyuer2019** [Percylevent/ROMA_TensortRT#2]
(roma) [root@localhost pytorch]# python extra.py
Using device: cuda
Loading full RoMa model...
/data/anaconda3/envs/roma/lib/python3.9/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/data/anaconda3/envs/roma/lib/python3.9/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=None`.
  warnings.warn(msg)
⚡ Fusing BatchNormalization layers for TensorRT optimization...
   - Fusing ConvRefiner scale 16
   - Fusing ConvRefiner scale 8
   - Fusing ConvRefiner scale 4
   - Fusing ConvRefiner scale 2
   - Fusing ConvRefiner scale 1
Full RoMa model loaded successfully.
Instantiating ONNX wrapper for core computation...
Exporting core RoMa model to ONNX at ./onnx/roma_core.onnx...
/data/ROMA_TensortRT-main/pytorch/extra.py:76: DeprecationWarning: You are using the legacy TorchScript-based ONNX export. Starting in PyTorch 2.9, the new torch.export-based ONNX exporter will be the default. To switch now, set dynamo=True in torch.onnx.export. This new exporter supports features like exporting LLMs with DynamicCache. We encourage you to try it and share feedback to help improve the experience. Learn more about the new export logic: https://pytorch.org/docs/stable/onnx_dynamo.html. For exporting control flow: https://pytorch.org/tutorials/beginner/onnx/export_control_flow_model_to_onnx_tutorial.html.
  torch.onnx.export(
/data/ROMA_TensortRT-main/pytorch/dino.py:237: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert (
/data/ROMA_TensortRT-main/pytorch/dino.py:240: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert (
/data/ROMA_TensortRT-main/pytorch/dino.py:420: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if npatch == N and w == h:
/data/ROMA_TensortRT-main/pytorch/dino.py:434: TracerWarning: Converting a tensor to a Python float might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  int(math.sqrt(N)),
/data/ROMA_TensortRT-main/pytorch/dino.py:435: TracerWarning: Converting a tensor to a Python float might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  int(math.sqrt(N)),
/data/ROMA_TensortRT-main/pytorch/dino.py:438: TracerWarning: Converting a tensor to a Python float might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  bb = (w0 / math.sqrt(N), h0 / math.sqrt(N))
/data/ROMA_TensortRT-main/pytorch/dino.py:441: TracerWarning: Converting a tensor to a Python number might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  cc = (bb[0].item(), bb[1].item()) # **** 1.make cc from tuple(tensor[float], tensor(float)) to tuple(float, float)
/data/ROMA_TensortRT-main/pytorch/dino.py:450: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  int(w0) == patch_pos_embed.shape[-2]
/data/ROMA_TensortRT-main/pytorch/dino.py:449: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert (
/data/ROMA_TensortRT-main/pytorch/dino.py:451: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  and int(h0) == patch_pos_embed.shape[-1]
/data/anaconda3/envs/roma/lib/python3.9/site-packages/torch/functional.py:554: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at /pytorch/aten/src/ATen/native/TensorShape.cpp:4322.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
/data/ROMA_TensortRT-main/pytorch/roma.py:1005: TracerWarning: Converting a tensor to a Python float might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  res = round(math.sqrt(C))
✅ ONNX export successful!

--- Verifying the exported ONNX model ---
2026-02-24 20:38:49.139439317 [W:onnxruntime:, transformer_memcpy.cc:74 ApplyImpl] 25 Memcpy nodes are added to the graph main_graph for CUDAExecutionProvider. It might have negative impact on performance (including unable to run CUDA graph). Set session_options.log_severity_level=1 to see the detail logs before this message.
❌ An error occurred during ONNX verification: 
Not equal to tolerance rtol=0.001, atol=0.0001

Mismatched elements: 585803 / 1016064 (57.7%)
Max absolute difference among violations: 2.95316165
Max relative difference among violations: 18108.42145083
 ACTUAL: array([[[[-0.973377, -0.973236, -0.973181, ...,  0.991528,  0.99458 ,
           0.996016],
         [-0.973628, -0.973762, -0.973178, ...,  0.992758,  0.995463,...
 DESIRED: array([[[[-0.973395, -0.973253, -0.973199, ...,  0.992153,  0.995016,
           0.996385],
         [-0.973644, -0.973776, -0.973189, ...,  0.993344,  0.995939,...

python版本为3.9
Package                  Version
------------------------ ---------
coloredlogs              15.0.1
einops                   0.8.2
filelock                 3.19.1
flatbuffers              25.12.19
fsspec                   2025.10.0
humanfriendly            10.0
importlib_metadata       8.7.1
Jinja2                   3.1.6
markdown-it-py           3.0.0
MarkupSafe               3.0.3
mdurl                    0.1.2
ml_dtypes                0.5.4
mpmath                   1.3.0
networkx                 3.2.1
numpy                    2.0.2
nvidia-cublas-cu12       12.8.4.1
nvidia-cuda-cupti-cu12   12.8.90
nvidia-cuda-nvrtc-cu12   12.8.93
nvidia-cuda-runtime-cu12 12.8.90
nvidia-cudnn-cu12        9.10.2.21
nvidia-cufft-cu12        11.3.3.83
nvidia-cufile-cu12       1.13.1.3
nvidia-curand-cu12       10.3.9.90
nvidia-cusolver-cu12     11.7.3.90
nvidia-cusparse-cu12     12.5.8.93
nvidia-cusparselt-cu12   0.7.1
nvidia-nccl-cu12         2.27.3
nvidia-nvjitlink-cu12    12.8.93
nvidia-nvtx-cu12         12.8.90
onnx                     1.19.1
onnx-simplifier          0.5.0
onnxruntime-gpu          1.19.2
onnxsim                  0.4.36
opencv-python            4.13.0.92
packaging                26.0
pillow                   11.3.0
pip                      25.1.1
protobuf                 6.33.5
Pygments                 2.19.2
rich                     14.3.3
setuptools               80.9.0
sympy                    1.14.0
torch                    2.8.0
torchvision              0.23.0
triton                   3.4.0
typing_extensions        4.15.0
wheel                    0.45.1
zipp                     3.23.0
